lib/rubycritic/commands/compare.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+require 'rubycritic/source_control_systems/base'
+require 'rubycritic/analysers_runner'
+require 'rubycritic/revision_comparator'
+require 'rubycritic/reporter'
+require 'rubycritic/commands/base'
+require 'rubycritic/commands/default'
+
+module RubyCritic
+  module Command
+    class Compare < Default
+      def initialize(options)
+        super
+        @build_number = 0
+      end
+
+      def execute
+        Config.no_browser = true
+        compare_branches
+        status_reporter.score = Config.feature_branch_score
+        status_reporter
+      end
+
+      private
+
+      attr_reader :paths, :status_reporter
+
+      def compare_branches
+        update_build_number
+        set_root_paths
+        analyse_branch(:base_branch)
+        analyse_branch(:feature_branch)
+        analyse_modified_files
+        compare_code_quality
+      end
+
+      # keep track of the number of builds and
+      # use this build number to create seperate directory for each build
+      def update_build_number
+        build_file_location = '/tmp/build_count.txt'
+        File.new(build_file_location, 'a') unless File.exist?(build_file_location)
+        @build_number = File.open(build_file_location).readlines.first.to_i + 1
+        File.write(build_file_location, @build_number)
+      end
+
+      def set_root_paths
+        Config.base_root_directory = Pathname.new(branch_directory(:base_branch))
+        Config.feature_root_directory = Pathname.new(branch_directory(:feature_branch))
+        Config.build_root_directory = Pathname.new(build_directory)
```

Comment:
This feature was developed with the existing `CI mode` in mind. Hoping this would eventually  become `CI mode` (since this is what CI mode is expected to do if I am not wrong).
Hence the term `build`.